src/elements/element.rectangle.js

```diff
@@ -50,11 +50,106 @@ module.exports = function(Chart) {
+					top: ceiling(vm.borderWidth[0], barSizeTopBottom),
+					right: ceiling(vm.borderWidth[0], barSizeLeftRight),
+					bottom: ceiling(vm.borderWidth[0], barSizeTopBottom),
+					left: ceiling(vm.borderWidth[0], barSizeLeftRight)
+				} :
+				vm.borderWidth instanceof Array && vm.borderWidth.length === 2 ? {
+					top: ceiling(vm.borderWidth[0], barSizeTopBottom),
+					right: ceiling(vm.borderWidth[1], barSizeLeftRight),
+					bottom: ceiling(vm.borderWidth[0], barSizeTopBottom),
+					left: ceiling(vm.borderWidth[1], barSizeLeftRight)
+				} :
+				vm.borderWidth instanceof Array ? {
+					top: ceiling(vm.borderWidth[0], barSizeTopBottom),
+					right: ceiling(vm.borderWidth[1], barSizeLeftRight),
+					bottom: ceiling(vm.borderWidth[2], barSizeTopBottom),
+					left: ceiling(vm.borderWidth[3], barSizeLeftRight)
+				} :
+				{
+					top: ceiling(vm.borderWidth, barSizeTopBottom),
+					right: ceiling(vm.borderWidth, barSizeLeftRight),
+					bottom: ceiling(vm.borderWidth, barSizeTopBottom),
+					left: ceiling(vm.borderWidth, barSizeLeftRight)
+				};
+		},
+		drawShape: function(borders) {
+			// Draw the inside shape...
 			var ctx = this._chart.ctx;
 			var vm = this._view;
-			var left, right, top, bottom, signX, signY, borderSkipped;
-			var borderWidth = vm.borderWidth;
+
+			ctx.beginPath();
+			ctx.fillStyle = vm.backgroundColor;
+			ctx.strokeStyle = 'rgba(0,0,0,0)';
+			ctx.lineWidth = 0;
+			ctx.moveTo(borders.bottom[0][0], borders.bottom[0][1]);
+			for (var i in borders) {
+				if (borders.hasOwnProperty(i)) {
+					ctx.lineTo(borders[i][1][0], borders[i][1][1]);
+				}
+			}
+			ctx.lineTo(borders.bottom[0][0], borders.bottom[0][1]);
+			ctx.fill();
+		},
+		adjustBorders: function(borders, borderWidth, borderSkipped) {
+			// Make sure border lengths are adjusted for border width.
+			if (!borderSkipped.includes('top') && borderWidth.top > 0) {
+				borders.left = [[borders.left[0][0], borders.left[0][1] - borderWidth.top/2], borders.left[1]];
+				borders.right = [borders.right[0], [borders.right[1][0], borders.right[1][1] - borderWidth.top/2]];
+			}
+			if (!borderSkipped.includes('bottom') && borderWidth.bottom > 0) {
+				borders.left = [borders.left[0], [borders.left[1][0], borders.left[1][1] + borderWidth.bottom/2]];
+				borders.right = [[borders.right[0][0], borders.right[0][1] + borderWidth.bottom/2], borders.right[1]];
+			}
+		},
+		drawBorders: function(borders, borderWidth, borderSkipped) {
+			// Draw the borders. Since canvas doesn't allow drawing the borders with
+			// different widths, draw each line individually.
+			var ctx = this._chart.ctx;
+			var vm = this._view;
+
+			ctx.strokeStyle = vm.borderColor;
+			for (var i in borders) {
+				if (borders.hasOwnProperty(i)) {
+					var border = borders[i];
+					ctx.lineWidth = borderWidth[i];
+					if (borderSkipped.includes(i) || !ctx.lineWidth || ctx.lineWidth === 0) {
+						// Skip drawing the border
+						continue;
+					}
+					ctx.beginPath();
```

Comment:
Since the `strokeStyle` doesn't change for each border, `beginPath()` and `stroke()` should be moved outside the loop:

```javascript
ctx.strokeStyle = vm.borderColor;
ctx.beginPath();
for (...) {
    ctx.moveTo(...)
    ctx.lineTo(...)
}
ctx.stroke();
```